packages/ppx_open/ppx_open.0.1/opam

```diff
@@ -0,0 +1,41 @@
+opam-version: "2.0"
+synopsis: "Idiomatic selective `open`s in OCaml"
+description:
+  "A ppx rewriter that provides idiomatic selective `open`s in OCaml."
+maintainer: ["ajo41@cam.ac.uk"]
+authors: ["Alistair O'Brien"]
+license: "MIT"
+homepage: "https://github.com/johnyob/ppx-open"
+bug-reports: "https://github.com/johnyob/ppx-open/issues"
+depends: [
+  "ocaml" {= "4.12.0"}
```

Comment:
Why is this required?

Reply:
If you're not compatible with OCaml 4.13 then this is strictly better to enable use of the ppx_open in a possible future 4.12.1 or 4.12.2 release:
```suggestion
  "ocaml" {>= "4.12.0" & < "4.13.0"}
```

Reply:
The ppx relies heavily on compiler typing internals. These change with each compiler release hence the `= 4.12.0` constraint.

Reply:
> These change with each compiler release hence the `= 4.12.0` constraint.

I don't think I've seen a patch release (4.12.0 vs 4.12.1) having any change to the typing API interface before. Do you have an example?